WordPress/Classes/ViewRelated/Blog/Site Settings/SiteSettingsViewController+Swift.swift

```diff
@@ -168,3 +168,233 @@ extension SiteSettingsViewController {
     fileprivate var ampSupportURL: String { return "https://support.wordpress.com/amp-accelerated-mobile-pages/" }
 
 }
+
+// MARK: - General Settings Table Section Management
+
+extension SiteSettingsViewController {
+
+    enum GeneralSettingsRow {
+        case title
+        case tagline
+        case url
+        case privacy
+        case language
+        case timezone
+        case bloggingReminders
+    }
+
+    func generalSettingsVisibleRows() -> [GeneralSettingsRow] {
+        var visibleRows: [GeneralSettingsRow] = [.title, .tagline, .url]
+
+        if blog.supportsSiteManagementServices() {
+            visibleRows.append(contentsOf: [.privacy, .language])
+        }
+
+        if blog.supports(.wpComRESTAPI) {
+            visibleRows.append(.timezone)
+        }
+
+        if Feature.enabled(.bloggingReminders) {
+            visibleRows.append(.bloggingReminders)
+        }
+
+        return visibleRows
+    }
+
+    @objc
+    func generalSettingsVisibleRowCount() -> Int {
```

Comment:
I wanted to hint that it was better to cache the value, rather than use it multiple times directly, but I believe in most scenarios this value won't be used multiple times in a row anyway.

Reply:
Resolved.  Did the same for `generalSettingsRows()`.